core/CronArchive/QueueConsumer.php

```diff
@@ -229,6 +227,18 @@ public function getNextArchivesToProcess()
                 continue;
             }
 
+            $alreadyInProgressId = $this->model->isArchiveAlreadyInProgress($invalidatedArchive);
+            if ($alreadyInProgressId) {
```

Comment:
btw I now wonder if this would actually work because we order the invalidations by `ORDER BY date1 DESC, period ASC, CHAR_LENGTH(name) ASC, idinvalidation ASC`. Doesn't this mean if there are like 5 queued for the same period, that it would pick the one with the lowest ID? so it wouldn't delete the other ones maybe because they are all lower?

On the other side we can't use the highest ID because it would maybe remove too many.

eg 
- invalidation 3: month sep 2020 
- invalidation 4: day sep 2020 14
- invalidation 5: month sep 2020 

eg if it was to select invalidation 5 first (and thus ignore and delete invalidation 3), and it archives month, and then it archives invalidation 4 the day  period, then the month would maybe never be rearchived.

Not sure it's clear what I mean but selecting the highest ID maybe wouldn't work either and could generally lead to maybe wrong data. 

Just wanted to check in if that's actually an issue that we select the lowest ID and then this would maybe not work. Not sure though might not be an issue

Reply:
@tsteur if this is about keeping different machines/processes from doing the same archiving job, then it should be fine? One process starts 3 because it's the earliest idinvalidation. Another picks up 5 and ignores it because 3 is in progress, but since 5 > 3, it's kept in the table.

Reply:
> Doesn't this mean if there are like 5 queued for the same period, that it would pick the one with the lowest ID? so it wouldn't delete the other ones maybe because they are all lower?

Ok, understanding this better... we could probably do idinvalidation DESC... will think a bit about whether that would be an issue...

Reply:
yeah the issue be that when

```
invalidation 3: month sep 2020
invalidation 4: month sep 2020
invalidation 5: day sep 2020 14
invalidation 6: month sep 2020
```

it works on 3, a parallel process would pick 4. It would notice it's higher than 3 and skip it but not delete it. So it then never actually deletes any of the duplicate ones. On the other side we also cannot fetch `idinvalidation desc` because we need to make sure after invalidation 5 the month will be reprocessed again.

The only way to prevent this is not to do a bulk insert and instead before the insert delete any already existing entry for the same period but not even sure if this could have any edge cases/race conditions (performance wise it should be fine considering there's an index on `idsite,date1,period` but would of course result in an additional query for each insert which isn't exactly great but maybe needed not sure). Although I reckon it should work because things are sorted by `ORDER BY date1 DESC, period ASC` and it should always archive first any lower invalidated period before archiving the higher period.

Reply:
> eg if it was to select invalidation 5 first (and thus ignore and delete invalidation 3), and it archives month, and then it archives invalidation 4 the day period, then the month would maybe never be rearchived.

so in this scenario:

* invalidation 4 (day) would be found first and archived (we order by period asc too)
* then invalidation 5, (we'd perhaps delete all invalidations below 5 just after starting it)

then we're fine. I guess it's more complicated when invalidations are added in between archiving. For example:

* invalidation 4 (day)
* start on invalidation 5
* meanwhile an invalidation for sep 14 is added back
* after invalidation 5 is done, we see sep 14 and re-archive
* the month is no longer accurate

Of course this would only be an issue if somehow sep 14 is invalidated, but no parent periods as well... which isn't the case unless there's a bug. In this case it would still work I think.

Will try w/ idinvalidation DESC since it definitely won't work right w/ idinvalidation ASC.

Reply:
👍  it probably works with `DESC` since it should archive any lower period first anyway

Reply:
Another scenario:

* invalidation 5 is started
* new month invalidation comes in
* we finish 5, see new month one. it's too soon since the last archive so we don't start it, but we don't remove it

this seems ok to me too?